inlineplz/linters/__init__.py

```diff
@@ -662,268 +657,289 @@ def vendored_path(path):
-    path = path or os.getcwd()
-    # http://stackoverflow.com/a/2186565
-    paths = set()
-    for root, dirnames, filenames in os.walk(path, topdown=True):
+    async def performance_hacks(self):
+        # https://github.com/npm/npm/issues/11283
+        # npm's progress bar makes npm installs much slower
         try:
-            for ignore in ignore_paths:
-                dirnames.remove(ignore)
-        except ValueError:
+            await self.run_command(["npm", "set", "progress=false"])
+        except Exception:
             pass
-        if should_ignore_path(root, ignore_paths):
-            continue
 
-        for filename in filenames:
-            full_path = os.path.join(root, filename)
-            if "text" in identify.tags_from_path(full_path):
-                paths.add(full_path)
-    return paths
+    @staticmethod
+    def cleanup():
+        """Delete standard installation directories."""
+        for install_dir in INSTALL_DIRS:
+            try:
+                shutil.rmtree(install_dir, ignore_errors=True)
+            except Exception:
+                print(traceback.format_exc())
+                print("Failed to delete {}".format(install_dir))
+
+    def should_ignore_path(self, path):
+        for ignore_path in self.ignore_paths:
+            if (
+                os.path.relpath(path).startswith(ignore_path)
+                or path.startswith(ignore_path)
+                or fnmatch.fnmatch(path, ignore_path)
+                or ignore_path in path.split(os.path.sep)
+            ):
+                return True
 
+        return False
 
-def should_autorun(config, filenames):
-    patterns = PATTERNS.get(config.get("language"))
-    if config.get("autorun"):
+    async def run_per_file(self, config, path=None):
+        path = path or os.getcwd()
+        cmd = self.run_config(config, self.config_dir)
+        run_cmds = []
+        file_tasks = []
+        patterns = PATTERNS.get(config.get("language"))
+        # TODO: Remove concurrency setting
+        paths = self.all_filenames_in_dir(path=path)
         for pattern in patterns:
-            if fnmatch.filter(filenames, pattern):
-                return True
+            for filepath in fnmatch.filter(paths, pattern):
+                if "text" in identify.tags_from_path(filepath):
+                    run_cmds.append(cmd + [filepath])
+                    file_tasks.append(self.event_loop.create_task(self.run_command(run_cmds[-1], timeout=5)))
 
-    return False
+        output = []
+        for run_cmd, file_task in zip(run_cmds, file_tasks):
+            _, out = await file_task
 
+            # Return the filename and the output
+            output.append((run_cmd[-1], out.strip()))
 
-def dotfiles_exist(config, path=None):
-    path = path or os.getcwd()
-    return any(
-        dotfile.strip() in os.listdir(path) for dotfile in config.get("dotfiles")
-    )
+        return output
 
+    def linters_to_run(self):
```

Comment:
[inline-plz]: `prospector: mccabe: LinterRunner.linters_to_run is too complex (16) (MC0001)`